base/collection.go

```diff
@@ -1,1122 +0,0 @@
-/*
```

Comment:
I've rearranged stuff now so that `GocbV2Bucket` and all of the methods are still in `base/collection.go` temporarily in order to make the diff easier to see.

We can do another git rename for this file in a follow-up PR post-merge back to `bucket_gocb.go`